Source/Atomic/Core/Context.h

```diff
@@ -228,6 +228,27 @@ class ATOMIC_API Context : public RefCounted
     void AddGlobalEventListener(GlobalEventListener* listener) { globalEventListeners_.Push(listener); }
     void RemoveGlobalEventListener(GlobalEventListener* listener) { globalEventListeners_.Erase(globalEventListeners_.Find(listener)); }
 
+    WeakPtr<Engine> engine_;

```

Comment:
I wouldnt necessarily say getters are more consistent. Bunch of classes expose their members, even `Vector3` and friends, there certainly are some others too. I am torn between kosher (getters) and practical (pointers).

Using pointers would immediately mean less typing, 4 characters per call and we can immediately take advantage of intellisense where with getters in order to get to relevant intellisense suggestion we would need to type at least 4 characters ("GetE", for engine for example).

Using getters would protect clueless people from shooting themselves in the foot, but maybe this software is not for clueless people and they should be in charge of protecting themselves :)

Maybe compromise is possible? C# could have properties and it would be very idiomatic. C++ could stay with pointers and js could... no idea what people do on js.